roles/bridge/tasks/setup_zfs.yml

```diff
@@ -0,0 +1,60 @@
+---
+- name: Install ZFS utilities
+  ansible.builtin.apt:
+    name: zfsutils-linux
+    state: present
+    update_cache: true
+
+- name: Create directory for filesystem image
+  ansible.builtin.file:
+    path: "{{ bridge_zfs_image_dir }}"
+    state: directory
+    owner: root
+    group: root
+    mode: "0755"
+
+- name: Create a filesystem image of specified size
+  ansible.builtin.command:
+    cmd: fallocate -l {{ bridge_zfs_image_size }} {{ bridge_zfs_image_path }}
+    creates: "{{ bridge_zfs_image_path }}"
+
+- name: Check if loopback device exists
+  ansible.builtin.command:
+    cmd: losetup -j {{ bridge_zfs_image_path }}
+  register: loopback_exists
+  changed_when: false
+  failed_when: false
+
+- name: Set up loopback device
+  ansible.builtin.command:
+    cmd: losetup -f {{ bridge_zfs_image_path }}
+  when: loopback_exists.stdout == ""
+  changed_when: false
+
+- name: Retrieve loopback device
+  ansible.builtin.shell:
+    cmd: |
+      set -o pipefail
+      losetup -a | grep {{ bridge_zfs_image_path }} | cut -d: -f1
+  register: loop_device
+  changed_when: false
+
+- name: Create ZFS pool with mount point
+  ansible.builtin.command:
+    cmd: zpool create -m {{ bridge_zfs_pool_mount_point }} {{ bridge_zfs_pool_name }} {{ loop_device.stdout }}
+    creates: "{{ bridge_zfs_pool_mount_point }}"
```

Comment:
I see we have zfs ansible module, does it not server the purpose of what we wanna do ?

Reply:
it's doesn't work as we need

Reply:
ah okay